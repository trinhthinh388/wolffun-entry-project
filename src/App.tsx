import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { SizeProvider } from './contexts';
import { CommonLayout } from './layouts';
import { Provider } from 'react-redux';
import store from './redux/store';

// Pages
const MarketPlace = React.lazy(() => import('./pages/Marketplace'));
const Hero = React.lazy(() => import('./pages/HeroDetail'));

type AppProps = {
  [key: string]: any;
};

const App: React.FC<AppProps> = () => (
  <BrowserRouter>
    <SizeProvider>
      <Provider store={store}>
        <CommonLayout>
          <React.Suspense fallback={null}>
            <Routes>
              <Route path="/" element={<h1>HOME</h1>} />
              <Route path="/buy" element={<MarketPlace />} />
              <Route path="/hero" element={<Hero />} />
            </Routes>
          </React.Suspense>
        </CommonLayout>
      </Provider>
    </SizeProvider>
  </BrowserRouter>
);

export default App;
