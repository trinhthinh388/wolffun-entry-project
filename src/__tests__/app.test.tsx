import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

test('App should mount properly', () => {
  const appComponent = renderer.create(<App />);

  const tree = appComponent.toJSON();
  expect(tree).toMatchSnapshot();
});
