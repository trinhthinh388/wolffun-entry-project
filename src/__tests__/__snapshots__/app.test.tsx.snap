// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should mount properly 1`] = `null`;
